src/main/java/Duke/Command/Parser.java

```diff
@@ -0,0 +1,269 @@
+            return null;
+        } else {
+            parsedTime = LocalDate.of(year, month, day).atTime(hour, minute);
+        }
+
+
+        return parsedTime;
+    }
+
+    /**
+     * Returns a String which is a task info in a local save data.
+     * Noted: you must specify it as local data, otherwise it can go wrong.
+     *
+     * @return Task retrieved from save data.
+     */
+    public String getSaveTask() {
+        String task;
+        char taskType = message.charAt(0);
+        if (taskType == 'D' || taskType == 'E') {
+            task = message.substring(8, message.indexOf("|", 8) - 1);
+        } else {
+            task = message.substring(8);
+        }
+
+        return task;
+    }
+
+    /**
+     * Returns a String which is a time info in a local save data.
+     * Noted: you must specify it as local data, otherwise it can go wrong.
+     *
+     * @return Time retrieved from save data.
+     */
+    public String getSaveTime() {
+        String time;
+
+        char taskType = message.charAt(0);
+        if ((taskType == 'D' || taskType == 'E') && message.contains("/")) {
+            time = message.substring(message.lastIndexOf("|") + 2);
+        } else {
+            time = "";
+        }
+        return time;
+    }
+
+    /**
+     * Returns a String which is an operation type in a line of command.
+     *
+     * @return Operation type parsed from users' one line of command.
+     * @throws DukeException Throws when the operation type does not belong to any one of the types that
+     * duke can do.
+     */
+    public String getOperationType() throws DukeException {
+        String operationType;
+        if (message.contains(" ")) {
+            operationType = message.substring(0, message.indexOf(" "));
+        } else {
+            operationType = message;
+        }
+
+        //If the task type does not belong to the three types, throw an error.
+        TaskList.OperationType[] operationTypes = TaskList.OperationType.values();
+        for (TaskList.OperationType o : operationTypes) {
+            if (message.startsWith(o.toString())) {
+                return operationType;
+            }
+        }
+        throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+
+    }
+
+    /**
+     * Returns a String which is task info in a line of command.
+     *
+     * @return Operation type parsed from users' one line of command.
+     * @throws DukeException Throws when the task info cannot be retrieved from users' one line of command.
+     */
+    public String getTask() throws DukeException {
+        String task = "";
+
+        if (message.startsWith("deadline") || message.startsWith("event") || message.startsWith("todo")
+                || message.startsWith("find")) {
+            //Get Task description and time if it has it.
+            if (message.contains("/")) {
+                task = message.substring(message.indexOf(" ") + 1, message.indexOf("/") - 1);
```

Comment:
ok, by right i should only be commenting bout code practices but i really really think youd find using regex & placing the logic of parsing into each of the subtasks much easier.

By having so many methods reliant on so many tasks, its gonna be hard to modify it.
But if you abstract the logic and put the parseLine() static method inside each task, could always just change that when you change the syntax. just fyi :)

Reply:
Yea, that will be a main stuff I will have to fix later. Thanks so much for your comments!